lib/dfe/reference_data/tweaked_reference_list.rb

```diff
@@ -25,26 +25,28 @@ def initialize(base, overrides, schema = nil)
         super(schema)
         @base = base
         @overrides = overrides
-
-        @overridden_all = base.all_as_hash.clone
-
-        overrides.each_entry do |id, record|
-          if record.nil?
-            @overridden_all.delete(id)
-          elsif @overridden_all.key? id
-            old_record = @overridden_all[id]
-            @overridden_all[id] = old_record.merge(record)
-          else
-            @overridden_all[id] = record.merge({ id: id })
-          end
-        end
+        @overridden_all = nil # Computed on demand in all_as_hash
```

Comment:
This line is no longer necessary as it's only being used for memoizing and defaults to `nil`.